### ISSUE TITLE
soundness bug in LIA

### ISSUE THREAD

**Issue by OriginLabs-Iguernlala** [OCamlPro/alt-ergo#340]
This example has status sat, but Alt-Ergo says it's unsat.

[convert-jpg2gif-query-1197.smt2.zip](https://github.com/OCamlPro/alt-ergo/files/4943372/convert-jpg2gif-query-1197.smt2.zip)

**Comment by OriginLabs-Iguernlala** [comment]
minimal example that reproduces the bug:
```
(declare-fun a () Int)
(declare-fun b () Int)
(declare-fun c () Int)
(declare-fun d () Int)

(assert (= (+ a (* b (- 524288)) (* c (- 4294967296)) (- d)) (- 1024)))
(assert (<= a 0))
(assert (>= c 0))
(assert (>= b 0))
(assert (>= d 0))
(check-sat)
```

**Comment by OriginLabs-Iguernlala** [comment]
Even simpler:
```
(declare-fun a () Int)
(declare-fun b () Int)
(declare-fun c () Int)
(declare-fun d () Int)
(declare-fun p (Int Int Int Int) Bool)

(assert (p a b c d)) ; just here for ordering of variables
(assert (<= a 0))
(assert (>= c 0))
(assert (>= b 0))
(assert (>= (+ (* (- 16384) c) (* (- 2) b) a 1) 0))
(check-sat)
```

**Comment by OriginLabs-Iguernlala** [comment]
The original bug should be fixed by PR #342 

But there is still another bug (with the original example) if option `--max-split=-1` is activated
